### DISCUSSION TITLE
Declare variable in one template and reference in another

### DISCUSSION THREAD

**Discussion by agilejoe** [SilentVoid13/Templater#552] [Help]
I have a scenario where I would like to declare a variable in the base template and reference it in the included template. Is this possible?

```
base template
---
let state = X

<%~ tp.file.include("[[front matter --template]]") %>

front matter -- template
--- 

if state == X ... 

```

**Comment by AB1908** [comment]
Hmm, I think the CustomJS plugin does this stuff better. Can also do some of this with User Scripts as well. Could you describe your use case a bit more?
